### PR TITLE
Personal/xiatia/additional parameter for api response

### PR DESCRIPTION
**Pull request by ACMoretxj**
## Description

* Added two extra parameters(`unusedSegments` and `invalidAccess`) in the query string for HL7V2 conversion for customizing response properties;
* Updated the documentation about API summary and data conversion;
* Fixed the document detail errors before: `/api/convert/{template}` -> `/api/convert/{srcDataType}/{template}`;

## Related issues

Addresses [issue #60 ].

## Testing

Added several extra unittests. All the UTs and E2E tests were passed.

### CODE REVIEW

File: docs/api-summary.md [hunk 1]
```diff
@@ -23,5 +23,5 @@ The API is secured using an API Key.
...
-|POST      |/api/convert/{template}    |Takes data and converts to FHIR using the {template} that is stored on the server.|
+|POST      |/api/convert/{srcDataType} |Takes data, and temporary templates as input and outputs FHIR data after applying the templates on the data. The entry-point template is passed base64-encoded in templateBase64 parameter, whereas other overriding templates are passed in the templatesOverrideBase64 parameter.  templatesOverrideBase64 is a base64-encoded json object containing map between the template name and the template content. You can also set `unusedSegments` and `invalidAccess` parameters in the query string to fetch UI-related segments.|
+|POST      |/api/convert/{srcDataType}/{template}    |Takes data and converts to FHIR using the {template} that is stored on the server. You can also set `unusedSegments` and `invalidAccess` parameters in the query string to fetch UI-related segments.|
```
Comment (ranvijaykumar) [inline docs/api-summary.md:27]: {srcDataType} is part of the template name itself. Therefore, we need not include {srcDataType} here. See the output of {{converterbaseURL}}/api/templates to get the names of all the templates.

Comment (ACMoretxj) [reply docs/api-summary.md:27]: Got it, I'll remove the extra {srcDataType} path parameter.

File: docs/api-summary.md [hunk 2]
```diff
@@ -23,5 +23,5 @@ The API is secured using an API Key.
...
-|POST      |/api/convert/{srcDataType} |Takes data, and temporary templates as input and outputs FHIR data after applying the templates on the data. The entry-point template is passed base64-encoded in templateBase64 parameter, whereas other overriding templates are passed in the templatesOverrideBase64 parameter.  templatesOverrideBase64 is a base64-encoded json object containing map between the template name and the template content.|
-|POST      |/api/convert/{template}    |Takes data and converts to FHIR using the {template} that is stored on the server.|
+|POST      |/api/convert/{srcDataType} |Takes data, and temporary templates as input and outputs FHIR data after applying the templates on the data. The entry-point template is passed base64-encoded in templateBase64 parameter, whereas other overriding templates are passed in the templatesOverrideBase64 parameter.  templatesOverrideBase64 is a base64-encoded json object containing map between the template name and the template content. You can also set `unusedSegments` and `invalidAccess` parameters in the query string to fetch UI-related segments.|
```
Comment (BoyaWu10) [inline docs/api-summary.md:26]: Will we request unusesSegments or invalidAccess independently? If not, we can combine them together into one option like "traceInfo=true".

Comment (BoyaWu10) [reply docs/api-summary.md:26]: The good point is that if we are going to add more trace or debug information in the future, we do not need to more "xxx=true" segments in the URL, and we can update the names of existing fields flexibly.

Comment (ACMoretxj) [reply docs/api-summary.md:26]: @ranvijaykumar What kind of implementation do you think is more appropriate? Use one parameter(like `traceInfo`) or multiple separate parameters(like `unusedSegments` and `invalidAccess` and so on)?

Comment (ranvijaykumar) [reply docs/api-summary.md:26]: unusedSegment is useful during template editing, while invalidAccess can be useful in production. Let's keep these separate.

Comment (BoyaWu10) [reply docs/api-summary.md:26]: Got it. Thanks!

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**